bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutationGCJClient.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc.async;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.core.IBulkMutation;
+import com.google.cloud.bigtable.data.v2.models.BulkMutationBatcher;
+import com.google.cloud.bigtable.data.v2.models.BulkMutationBatcher.BulkMutationFailure;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.cloud.bigtable.util.ApiFutureUtil;
+import com.google.common.base.Preconditions;
+
+/**
+ * This class is meant to replicate existing {@link BulkMutation} while translating calls to
+ * Google-Cloud-Java's {@link BulkMutationBatcher} api.
+ */
+public class BulkMutationGCJClient implements IBulkMutation {
```

Comment:
The ordering of the methods should be in order:

1. constructor
2. add
3. sendUnsent
4. flush
5. isFlushed